src/components/ToastTextToCad.tsx

```diff
@@ -516,8 +519,20 @@ export function ToastPromptToEditCadSuccess({
             name={'Reject'}
             onClick={() => {
               sendTelemetry(modelId, 'rejected', token).catch(reportRejection)
-              codeManager.updateCodeEditor(oldCode)
-              kclManager.executeCode().catch(reportRejection)
+              // revert to before the prompt-to-edit
+              if (isDesktop()) {
+                for (const file of oldFiles) {
+                  if (file.type !== 'kcl') {
+                    // only need to write the kcl files
+                    // as the endpoint would have never overwritten other file types
+                    continue
+                  }
+                  window.electron.writeFile(file.absPath, file.fileContents)
+                }
+              } else {
+                codeManager.updateCodeEditor(oldCode)
+                kclManager.executeCode().catch(reportRejection)
+              }
```

Comment:
Shouldn't this be also part of SystemIO? At least the isDesktop() part that's directly writing to files?

cc @nadr0